res/controllers/Denon-Prime-4-scripts.js

```diff
@@ -0,0 +1,921 @@
+            this.inputWheel(channel, control, (this.wheelMSB << 7) + value, status, group);
+        },
+        previousPosition: null,
+        wrappingValue: Math.pow(2, 14),
+        relativeFromAbsolute: function(value) {
+            // The first value of the controller will probably be random
+            // and thus we just have to swallow it until we have the second value
+            // to find the difference
+            if (this.previousPosition === null) {
+                this.previousPosition = value;
+                return 0;
+            }
+            // This finds the shortest distance between the current value
+            // and the last one, and preserves the orientation
+            const delta = value - this.previousPosition;
+            let remainder = ((delta % this.wrappingValue) + this.wrappingValue) % this.wrappingValue;
+            if (remainder * 2 > this.wrappingValue) {
+                remainder -= this.wrappingValue;
+            }
+            this.previousPosition = value;
+            return remainder;
+        },
+        jogScale: function(val) {
+            // wheelSensitivity is user-configurable, see top of file.
+            return val * wheelSensitivity;
+        },
+        inputWheel: function(channel, control, value, _status, _group) {
+            value = this.relativeFromAbsolute(value);
+            if (engine.isScratching(this.deck)) {
+                engine.scratchTick(this.deck, value);
+            } else {
+                this.inSetValue(this.jogScale(value));
+            }
+        },
+    });
+
+    // Jog Wheel LED
+    this.jogWheelLed = new components.Component({
+        midi: [0x90 + midiChannel, 0x21],
+        on: colDeck[deckNumbers - 1],
+        trigger: function() {
+            this.send(this.on);
+        },
+    });
+
+    // Slip Mode Button
+    this.slipButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x24],
+        key: "slip_enabled",
+        type: components.Button.prototype.types.toggle,
+    });
+
+    // Loop In Button
+    this.loopInButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x25],
+        key: "loop_in",
+    });
+
+    // Loop Out Button
+    this.loopOutButton = new components.Button({
+        midi: [0x90 + midiChannel, 0x26],
+        key: "loop_out",
+    });
+
+    // Load Buttons
+    this.deckLoad = new components.Button({
+        midi: [0x9F, midiChannel - 3],
+        key: "LoadSelectedTrack",
+
+        // TODO: Make load buttons reflect colour of currently active deck.
+        //       This is harder than expected due to the fact that I rely on the MIDI channel
+        //       of each physical deck component to determine which deck it's manipulating in
+        //       Mixxx. The deck load buttons, however, are both on MIDI channel 16, so I need
+        //       a better way to determine this.
+        off: Prime4.rgbCode.whiteDark,
+        on: Prime4.rgbCode.white,
```

Comment:
Since we made this part of the deck, we can just base it of the deck. same as the jogwheelLED. right?

```suggestion
        off: colDeckDark[deckNumbers - 1],
        on: colDeck[deckNumbers - 1],
```